azure-kit-starter/src/main/java/com/vaadin/azure/starter/ui/ClusterSupport.java

```diff
@@ -0,0 +1,245 @@
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * Cluster support for Vaadin applications. This component allows receiving
+ * events from the cluster running the Vaadin application.
+ */
+public class ClusterSupport implements VaadinServiceInitListener {
+
+    private static final long serialVersionUID = 1L;
+
+    /** Configuration property name. */
+    public static final String PROPERTY_APP_VERSION = "app.version";
+    public static final String ENV_APP_VERSION = "APP_VERSION";
+
+    /* Control cookie name. */
+    public static final String CURRENT_VERSION_COOKIE = "app-version";
+    public static final String UPDATE_VERSION_COOKIE = "app-update";
+    public static final String STICKY_CLUSTER_COOKIE = "INGRESSCOOKIE";
+
+    /* Global listener for version changes. */
+    private SwitchVersionListener switchVersionListener;
+
+    private String appVersion;
+
+    @Tag("version-notificator")
+    @JsModule("./components/version-notificator.ts")
+    public static class VersionNotificator extends Component {
+        private static final long serialVersionUID = 1L;
+
+        VersionNotificator(String current, String update) {
+            getElement().setProperty("currentVersion", current);
+            getElement().setProperty("updateVersion", update);
+        }
+
+        public Registration addSwitchVersionEvent(
+                ComponentEventListener<SwitchVersionEvent> listener) {
+            getLogger().debug("Adding listener for SwitchVersionEvent.");
+            return addListener(SwitchVersionEvent.class, listener);
+        }
+
+        @DomEvent("load-version")
+        public static class SwitchVersionEvent
+                extends ComponentEvent<VersionNotificator> {
+            private static final long serialVersionUID = 1L;
+
+            public SwitchVersionEvent(VersionNotificator source,
+                    boolean fromClient) {
+                super(source, fromClient);
+            }
+        }
+    }
+
+    /**
+     * Interface for receiving events on cluster change.
+     */
+    public interface SwitchVersionListener extends Serializable {
+
+        /**
+         * Notify about the cluster node change to allow graceful transition of
+         * the users. <b>Note:</b> Even returning <code>false</code>, the
+         * application might still be shut down by the environment.
+         *
+         * @param vaadinRequest
+         *            Vaadin request when the change is initiated.
+         * @param vaadinResponse
+         *            Response from the server to be sent to the client.
+         * @return <code>@code true</code> if the cluster change is ok, false if
+         *         the change should not be performed.
+         */
+        boolean clusterSwitch(VaadinRequest vaadinRequest,
+                VaadinResponse vaadinResponse);
+    }
+
+    /**
+     * Register the version switch listener.
+     *
+     * @param listener
+     *            SwitchVersionListener to register.
```

Comment:
The type is already part of the signature, no need to further specify it.
```suggestion
     *            the listener to register.
```

Reply:
Also, better document also what happens if set to `null`.

Reply:
Updated the JavaDoc, documented the null case.